src/tools/ci/install_meson_and_rebase.sh

```diff
@@ -0,0 +1,86 @@
+#! /bin/sh
+
+set -e
+set -x
+
+MESON_REPO=$1
+MESON_BRANCH=$2
+CURRENT_OS=$3
+
+if [ $# -ne 3 ]; then
+    echo "install_meson_and_rebase.sh <MESON_REPO> <MESON_BRANCH> <OS>"
+    exit 1
+fi
+
+case $CURRENT_OS in
+    freebsd|linux|macos|windows|mingw)
+    ;;
+    *)
+        echo "unsupported operating system ${CURRENT_OS}"
+        exit 1
+    ;;
+esac
+
+# After repartition, hidden files are not copied. Copy them to
+# working dir
+if [ "$CURRENT_OS" = 'freebsd' ]; then
+    cp -r $CIRRUS_WORKING_DIR.orig/.[^.]* $CIRRUS_WORKING_DIR/
+    chown -R postgres:postgres .[^.]*
+fi
+
+# pip is not installed as a default on some OSes. First install
+# pip, then install meson by using pip.
+install_meson () {
+    case $CURRENT_OS in
+        linux)
+            apt update
+            apt install python3-pip -y
+            pip install git+${MESON_REPO}@${MESON_BRANCH}
+        ;;
+        freebsd)
+            python3 -m ensurepip --upgrade
+            pip3 install git+${MESON_REPO}@${MESON_BRANCH}
+        ;;
+        macos)
+            python3 -m site --user-base && \
+                PATH="`python3 -m site --user-base`/bin:$PATH"
+            pip3 install git+${MESON_REPO}@${MESON_BRANCH}
+        ;;
+        mingw)
+            pacman -S --needed --noconfirm ucrt64/mingw-w64-ucrt-x86_64-python-pip
+            pip3 install git+${MESON_REPO}@${MESON_BRANCH}
+        ;;
+        windows)
+            pip install git+${MESON_REPO}@${MESON_BRANCH}
+        ;;
+    esac
+}
+
+# Rebase current branch onto Postgres HEAD
+rebase_onto_postgres () {
+    case $CURRENT_OS in
+        freebsd|linux)
+su postgres <<-EOF
+    git config user.email 'postgres-ci@example.com'
+    git config user.name 'Postgres CI'
+    git remote add default-postgres https://github.com/postgres/postgres.git
+    git fetch default-postgres master
+    git rebase --no-verify default-postgres/master
+EOF
+        ;;
+        windows|mingw|macos)
+            git config user.email 'postgres-ci@example.com'
+            git config user.name 'Postgres CI'
+            # There are file permission issues on Windows. Set filemode to
+            # false for windows, then reset
+            git config core.filemode false
```

Comment:
What are these issues?

Reply:
Chmod issues, for example:
```shell
diff --git a/config/install-sh b/config/install-sh
old mode 100755
new mode 100644
```

Reply:
Maybe change the comment to "windows looses the executable bit, causing an unnecessary diff".

And as I mentioned, I think this would look a lot better if just that  part were OS dependent.